server/apps/newsletter/tasks.py

```
@@ -43,4 +46,52 @@ def send_newsletter_via_email():
+        
+        subscriber.beehiiv_id = response.get('id')
+        subscriber.synced_to_beehiiv = True
+        subscriber.sync_error = None
+        subscriber.save()
+        
+        logger.info(f"Successfully synced subscriber {subscriber.email} to Beehiiv")
+        
+    except Exception as e:
+        logger.error(f"Failed to sync subscriber {subscriber_id} to Beehiiv: {str(e)}")
+        retry_countdown = 60 * (2 ** self.request.retries)  # Exponential backoff
+        raise self.retry(exc=e, countdown=retry_countdown)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)